python/lsst/pipe/tasks/assembleChi2Coadd.py

```diff
@@ -220,4 +223,97 @@ def run(self, inputCoadds: list[afwImage.Exposure]) -> pipeBase.Struct:
         # Create the exposure
         maskedImage = refExp.maskedImage.Factory(image, mask=mask, variance=variance)
         chi2coadd = refExp.Factory(maskedImage, exposureInfo=refExp.getInfo())
+        chi2coadd.info.setFilter(None)
         return pipeBase.Struct(chi2Coadd=chi2coadd)
+
+
+class DetectChi2SourcesConnections(
+    pipeBase.PipelineTaskConnections,
+    dimensions=("tract", "patch", "skymap"),
+    defaultTemplates={
+        "inputCoaddName": "deepChi2",
+        "outputCoaddName": "deepChi2"
+    }
+):
+    detectionSchema = cT.InitOutput(
+        doc="Schema of the detection catalog",
+        name="{outputCoaddName}Coadd_det_schema",
+        storageClass="SourceCatalog",
+    )
+    exposure = cT.Input(
+        doc="Exposure on which detections are to be performed",
+        name="{inputCoaddName}Chi2Coadd_calexp",
+        storageClass="ExposureF",
+        dimensions=("tract", "patch", "skymap")
+    )
+    outputSources = cT.Output(
+        doc="Detected sources catalog",
+        name="{outputCoaddName}Chi2Coadd_det",
+        storageClass="SourceCatalog",
+        dimensions=("tract", "patch", "skymap")
+    )
+
+
+class DetectChi2SourcesConfig(pipeBase.PipelineTaskConfig, pipelineConnections=DetectChi2SourcesConnections):
+    detection = pexConfig.ConfigurableField(
+        target=SourceDetectionTask,
+        doc="Detect sources in chi2 coadd"
+    )
+
+    idGenerator = SkyMapIdGeneratorConfig.make_field()
+
+    def setDefaults(self):
+        super().setDefaults()
+        self.detection.reEstimateBackground = False
+        self.detection.thresholdValue = 3
+
+
+class DetectChi2SourcesTask(pipeBase.PipelineTask):
+    _DefaultName = "detectChi2Sources"
+    ConfigClass = DetectChi2SourcesConfig
+
+    def __init__(self, schema=None, **kwargs):
+        # N.B. Super is used here to handle the multiple inheritance of PipelineTasks, the init tree
+        # call structure has been reviewed carefully to be sure super will work as intended.
+        super().__init__(**kwargs)
+        if schema is None:
+            schema = afwTable.SourceTable.makeMinimalSchema()
+        self.schema = schema
+        self.makeSubtask("detection", schema=self.schema)
+        self.detectionSchema = afwTable.SourceCatalog(self.schema)
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        inputs = butlerQC.get(inputRefs)
+        idGenerator = self.config.idGenerator.apply(butlerQC.quantum.dataId)
+        inputs["idFactory"] = idGenerator.make_table_id_factory()
+        inputs["expId"] = idGenerator.catalog_id
+        outputs = self.run(**inputs)
+        butlerQC.put(outputs, outputRefs)
+
+    def run(self, exposure: afwImage.Exposure, idFactory: afwTable.IdFactory, expId: int) -> pipeBase.Struct:
+        """Run detection on a chi2 exposure.
+
+        Parameters
+        ----------
+        exposure :
```

Comment:
exposure : `lsst.afw.image.Exposure`

Reply:
No need, since I used type hints in the signature.

Reply:
Ah! Does it appear automatically when you build the docs then? Amazing!